### PR TITLE
usysconf rewrite

### PR DESCRIPTION
**Pull request by chrisppy**
This is a complete rewrite of usysconf from C with hardcoded handlers to Go with TOML based config files.

### CODE REVIEW

**Comment by DataDrake** [comment]
♥ I'll look at this after work. Thank you so much for all of your hard work on this!

**Comment by chrisppy** [comment]
@DataDrake  I have finished with my cleanup and this is ready for review.

File: config.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+		}
+	}
+
+	return c
+}
+
+func (c *config) finish() {
+	for _, out := range c.Output {
+		t := time.Now()
+		now := fmt.Sprintf("%02d:%02d:%02d", t.Hour(), t.Minute(), t.Second())
```
Comment (JoshStrobl) [inline config.go]: Would recommend using `func (Time) Format` instead, easier to parse IMO.

File: config.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+		switch out.Status {
+		case skipped:
+			wlog.Infof("Skipped %s\n", out.Name)
+			fmt.Fprintf(os.Stdout, "\033[30;48;5;220m ⁓ \033[7m %s \033[27m %-42s \033[49;38;5;220m %s\033[0m\n", now, out.Name, out.SubTask)
+		case failure:
+			wlog.Errorf("Failure for %s due to %s\n", out.Name, out.Message)
+			fmt.Fprintf(os.Stdout, "\033[30;48;5;208m ✗ \033[7m %s \033[27m %-42s \033[49;38;5;208m %s\033[0m\n", now, out.Name, out.SubTask)
+		case success:
+			wlog.Goodln(out.Name)
+			fmt.Fprintf(os.Stdout, "\033[30;48;5;040m 🗸 \033[7m %s \033[27m %-42s \033[49;38;5;040m %s\033[0m\n", now, out.Name, out.SubTask)
```
Comment (JoshStrobl) [inline config.go]: Repetitive colors should just be defined as variables IMO or create various logging functions. Examples: https://github.com/JoshStrobl/trunk/blob/master/trunk.go

File: config.go [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+}
+
+func (c *content) skipProcessing() bool {
+	check := c.Check
+	if check != nil {
+		if err := check.checkPaths(); err != nil {
+			wlog.Errorln(err.Error())
+			return true
+		}
+	}
```
Comment (JoshStrobl) [inline config.go]: Should just use `c.Check` directly instead of setting it as a variable. Kinda needless.

File: config.go [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+	if *isForced {
+		return false
+	}
+
+	skip := c.Skip
+	if skip == nil {
```
Comment (JoshStrobl) [inline config.go]: Can just use c.Skip directly.

File: doc/usysconf.1.scd [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+usysconf will run a variety of triggers to ensure system consistency and health.
+
+For information on the how to run configuration files, see *usysconf-run*(1).
+
+# CONFIGURATION
+
+usysconf searches for config files in the following location, in this order:
+
+. /usr/share/defaults/usysconf.d (or whatever path is set for USRDIR)
+. /etc/usysconf.d (or whatever path is set for SYSDIR)
```
Comment (JoshStrobl) [inline doc/usysconf.1.scd]: This should really be inverse IMO. We should search /etc/usysconf.d, map out configs, then when they're not set, proceed to set the ones from USRDIR.

**Review by JoshStrobl — CHANGES_REQUESTED** [review]
Just an early review. Some other feedback:

1. Not entirely in agreement about the function naming scheme, namely they should be UpperCamelCase. Same goes for struct names (e.g. `check` to `Check`).
2. While I'm appreciative of the man pages dev docs you've created, I do not find it likely that they will be extensively utilized as opposed to just Markdown files (whether maintained on a usysconf GitHub wiki or placed in the Help Center).
3. Dire need of documentation / code commenting, including following Go documentation conventions for functions, see https://blog.golang.org/godoc

Will review the conf examples as soon as possible. Otherwise excellent work.

File: config.go (hunk 1 above)
Comment (chrisppy) [reply config.go]: Resolved.

File: config.go (hunk 3 above)
Comment (chrisppy) [reply config.go]: Resolved.

File: config.go (hunk 4 above)
Comment (chrisppy) [reply config.go]: Resolved.

File: doc/usysconf.1.scd (hunk 5 above)
Comment (chrisppy) [reply doc/usysconf.1.scd]: Resolved.

File: config.go (hunk 2 above)
Comment (chrisppy) [reply config.go]: Resolved.

**Comment by chrisppy** [comment]
@JoshStrobl @DataDrake  I have resolved the comments noted.

File: config.go [hunk 6]
```diff
@@ -164,20 +164,28 @@ func Load(name string) *Config {
 // Finish is the last function to be exexuted by any configuration to output
 // details to the user.
 func (c *Config) Finish() {
+	ansiYellow := "\033[30;48;5;226m"
```
Comment (DataDrake) [inline config.go]: Shouldn't these match the colors for Waterlog?

See: https://github.com/DataDrake/waterlog/blob/master/format/style.go

**Review by DataDrake — CHANGES_REQUESTED** [review]

File: config.go (hunk 6 above)
Comment (chrisppy) [reply config.go]: Colors have been fixed

Comment (chrisppy) [reply config.go]: Should the skipped configurations be warning or info lines?

**Comment by Jacalz** [comment]
You have done great work with this @chrisppy :+1:
Any news on getting this merged?  :)

**Comment by chrisppy** [comment]
Thanks @Jacalz 

@DataDrake or @JoshStrobl are there any more changes needed before merge?

**Comment by DataDrake** [comment]
@chrisppy not at the moment. The only reason we haven't merged this is having the time to sit down and integrate your changes into other packages. Trust me, we haven't forgotten!

**Comment by chrisppy** [comment]
Thanks @DataDrake

**Comment by EbonJaeger** [comment]
🎉 🎉 🎉

**Comment by Jacalz** [comment]
Epic 🎉

**Pull request merged**